client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -650,7 +667,8 @@ private ConcurrentHashMap<Integer, PartitionLocation> registerShuffleInternal(
       numRetries--;
     }
 
-    return null;
+    throw new CelebornIOException(
```

Comment:
Better not to wrapper CelebornIOException use another CelebornIOException, You can use ExceptionUtils.wrapAndThrowIOException to cast all exception to CelebornIOException